### PR TITLE
hermit init --idea

### PR DESCRIPTION
**Pull request by quad**

### CODE REVIEW

File: .idea/externalDependencies.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ExternalDependencies">
+    <plugin id="org.squareup.cash.hermit.idea-plugin" />
+  </component>
+</project>
```
Comment (quad) [inline .idea/externalDependencies.xml:6]: @alecthomas Should this go in `files/.idea/externalDependencies.xml`?

Comment (alecthomas) [reply .idea/externalDependencies.xml:6]: Yeah, though I think just `files/externalDependencies.xml` is okay.

File: env.go [hunk 2]
```diff
@@ -75,9 +75,10 @@ const (
 
 // Config for a Hermit environment.
 type Config struct {
-	Envars    envars.Envars `hcl:"env,optional" help:"Extra environment variables."`
-	Sources   []string      `hcl:"sources,optional" help:"Package manifest sources."`
-	ManageGit bool          `hcl:"manage-git,optional" default:"true" help:"Whether Hermit should automatically 'git add' new packages."`
+	Envars      envars.Envars `hcl:"env,optional" help:"Extra environment variables."`
+	Sources     []string      `hcl:"sources,optional" help:"Package manifest sources."`
+	ManageGit   bool          `hcl:"manage-git,optional" default:"true" help:"Whether Hermit should automatically 'git add' new packages."`
+	AddIJPlugin bool          `hcl:"idea,optional" default:"false" help:"Whether Hermit should automatically add the IntelliJ IDEA plugin."`
```
Comment (alecthomas) [inline env.go:81]: I don't think this one needs to exist as it's only used once at `init` time, which is too late for this to apply.

**Review by alecthomas — APPROVED** [review]
Great idea, thanks ssr!

File: env.go (hunk 2 above)
Comment (quad) [reply env.go:81]: How can `Init` get the configuration option if not from `Config`?

Comment (alecthomas) [reply env.go:81]: Yep true!

**Review by alecthomas — APPROVED** [review]

File: env.go [hunk 3]
```diff
@@ -124,6 +128,31 @@ func Init(l *ui.UI, env string, distURL string, stateDir string, config Config)
...
+				return errors.WithStack(err)
+			}
+
+			if useGit {
+				if err = util.RunInDir(b, env, "git", "add", "-f", extDepPath); err != nil {
+					return errors.WithStack(err)
+				}
+			}
+		} else {
+			l.Infof("IntelliJ IDEA configuration already exists; skipping adding configuration.")
```
Comment (quad) [inline env.go:151]: @alecthomas  Is this log OK?

File: it/full/spec/it_spec.sh [hunk 4]
```diff
@@ -14,6 +14,9 @@ Describe "Hermit"
     It "creates hermit activation script in ./bin/activate-hermit"
       The file ./bin/activate-hermit should be exist
     End
+    It "creates intellij idea plugin configuration in ./.idea/externalDependencies.xml"
+      The file ./.idea/externalDependencies.xml should be exist
+    End
```
Comment (quad) [inline it/full/spec/it_spec.sh:19]: @alecthomas Is this an OK place to add the integration test?

File: env.go (hunk 3 above)
Comment (alecthomas) [reply env.go:151]: Should be fine I think.

File: it/full/spec/it_spec.sh (hunk 4 above)
Comment (alecthomas) [reply it/full/spec/it_spec.sh:19]: Yep!

**Review by alecthomas — APPROVED** [review]

**Pull request merged**